python/letsql/backends/let/hotfix.py

```diff
@@ -0,0 +1,165 @@
+    frozen,
+)
+from attr.validators import (
+    instance_of,
+)
+
+import letsql
+from letsql.expr.relations import (
+    CachedNode,
+    replace_cache_table,
+)
+from letsql.common.utils.hotfix_utils import (
+    maybe_hotfix,
+    none_tokenized,
+)
+
+
+@frozen
+class LETSQLAccessor:
+    expr = field(validator=instance_of(ibis.expr.types.core.Expr))
+    node_types = (ops.DatabaseTable, ops.SQLQueryResult)
+
+    @property
+    def op(self):
+        return self.expr.op()
+
+    @property
+    def cached_nodes(self):
+        return self.op.find(CachedNode)
+
+    @property
+    def storage(self):
+        if self.is_cached:
+            return self.op.storage
+        else:
+            return None
+
+    @property
+    def storages(self):
+        return tuple(node.storage for node in self.cached_nodes)
+
+    @property
+    def _backends(self):
+        (backends, _) = self.expr._find_backends()
+        return backends
+
+    @property
+    def is_letsql(self):
+        names = set(backend.name for backend in self._backends)
+        if letsql.backends.let.Backend.name in names:
+            if len(names) > 1:
+                raise ValueError
+            else:
+                return True
+        else:
+            return False
+
+    @property
+    def ls_con(self):
+        if self.is_letsql:
+            (con,) = self._backends
+            return con
+        else:
+            return None
+
+    @property
+    def backends(self):
+        if self.is_letsql:
+            return tuple(
+                set((self.ls_con,) + tuple(dt.source for dt in self.native_dts))
+            )
+        else:
+            return self._backends
+
+    @property
+    def is_multiengine(self):
+        (_, *rest) = set(self.backends)
+        return bool(rest)
+
+    @property
+    def dts(self):
+        return self.op.find(self.node_types)
+
+    @property
+    def native_dts(self):
+        sources = self.ls_con._sources.sources
+        return tuple(sources[dt] for dt in self.dts)
+
+    @property
+    def native_expr(self):
+        _sources = self.ls_con._sources
+
+        def replace_table(n, _, **_kwargs):
+            return _sources.get_table_or_op(n, n.__recreate__(_kwargs))
+
+        return self.op.replace(replace_table).to_expr()
+
+    @property
+    def is_cached(self):
+        return isinstance(self.op, CachedNode)
+
+    @property
+    def has_cached(self):
+        return bool(tuple(self.op.find_topmost(CachedNode)))
+
+    @property
+    def uncached(self):
+        if self.has_cached:
+            return self.op.map_clear(replace_cache_table).to_expr()
+        else:
+            return self.expr
+
+    @property
+    def uncached_one(self):
+        if self.is_cached:
+            return self.op.parent.to_expr()
+        else:
+            return self.expr
+
+    def get_key(self):
+        if self.is_cached:
+            return self.storage.get_key(
+                self.ls_con._register_and_transform_cache_tables(
```

Comment:
Why the use of `_register_and_transform_cache_tables` in here?

Reply:
currently, the hash depends on all downstream cached nodes being replaced by their cached location